src/util/vscodePrompts.ts

```diff
@@ -0,0 +1,111 @@
+  const options: vscode.OpenDialogOptions = {
+    canSelectFiles: true,
+    canSelectFolders: false,
+    canSelectMany: false,
+    ...customOptions,
+  };
+
+  const result = await vscode.window.showOpenDialog(options);
+  if (result && result.length > 0) {
+    return result[0];
+  }
+
+  return undefined;
+}
+
+async function askForFolder(
+  customOptions: vscode.OpenDialogOptions
+): Promise<vscode.Uri | undefined> {
+  const options: vscode.OpenDialogOptions = {
+    canSelectFiles: false,
+    canSelectFolders: true,
+    canSelectMany: false,
+    ...customOptions,
+  };
+
+  const result = await vscode.window.showOpenDialog(options);
+  if (result && result.length > 0) {
+    return result[0];
+  }
+
+  return undefined;
+}
+
+export async function askForYamlInputMethod(): Promise<string | undefined> {
+  const defaultYamlChoice = vscode.workspace
+    .getConfiguration(EXTENSION_CONFIGURATION_IDENTIFIER)
+    .get<string>(DEFAULT_YAML_CHOICE_IDENTIFIER);
+
+  if (defaultYamlChoice) {
+    return defaultYamlChoice;
+  }
+
+  return vscode.window.showQuickPick([INPUT_YAML_OPTIONS.FROM_FILE, INPUT_YAML_OPTIONS.FROM_URL], {
+    ignoreFocusOut: true,
+    placeHolder: "Select one of the following choices.",
+  });
+}
+
+export async function askForYamlFile(): Promise<vscode.Uri | undefined> {
+  return askForFile({
+    openLabel: "Generate from this file",
+  });
+}
+
+export async function askForYamlURL(): Promise<string | undefined> {
+  const defaultYamlPath = vscode.workspace
+    .getConfiguration(EXTENSION_CONFIGURATION_IDENTIFIER)
+    .get<string>(DEFAULT_YAML_PATH_IDENTIFIER);
+
+  return vscode.window.showInputBox({
+    placeHolder: "e.g. com.example",
+    prompt: "Input yaml path for your project",
+    ignoreFocusOut: true,
+    value: defaultYamlPath,
+  });
+}
+
+export async function askForTargetFolder(defaultUri?: vscode.Uri): Promise<vscode.Uri | undefined> {
+  return askForFolder({
+    openLabel: "Generate into this package",
+    defaultUri: defaultUri,
+  });
+}
+
+export async function askForPackageName(srcDir: string): Promise<string | undefined> {
+  // looking for "/java/" from the path
+  let index = srcDir.toLowerCase().indexOf(path.sep + "java" + path.sep);
+  let defaultPackageName: string | undefined;
+  if (index > -1) {
+    // use everything after the "/java/" to be the package name
+    index = index + 6;
+    defaultPackageName = srcDir.substring(index).replace(new RegExp("\\" + path.sep, "g"), ".");
+  }
+
+  const packageNameRegex = /^[a-z][a-z0-9_]*(\.[a-z0-9_]+)*$/; // used to validate the package name
+  return await vscode.window.showInputBox({
+    placeHolder: "e.g. com.example",
+    prompt: "Input package name for your project",
+    ignoreFocusOut: true,
+    validateInput: (value: string) => {
+      if (packageNameRegex.test(value) === false) {
+        return "Invalid package name";
+      } else {
+        return null;
+      }
+    },
+    value: defaultPackageName,
```

Comment:
For now we should alter this method to allow user's to enter an empty package name.  This partly addresses #11